spec/unit/http/session_spec.rb

```diff
@@ -73,6 +73,37 @@ def resolve(session, name, ssl_context: nil)
         session.route_to(:westbound)
       }.to raise_error(ArgumentError, "Unknown service westbound")
     end
+
+    it 'routes to the service when given a puppet URL with an explicit host' do
+      service_url = "https://example.com:8140/puppet/v3"
+      fileserver = double('filerserver', connect: nil, url: service_url)
```

Comment:
You should be stubbing out the `Net::HTTP.start` method rather than our `connect` method

Reply:
I don't know what I was thinking here. I changed this to just return a FileServer object and stubbed the Net::HTTP start call